### PR TITLE
[Telegram] Support for formatted messages

### PR DESCRIPTION
**Pull request by thorstenbe**
This pull requests adds the ability to write messages that contain markdown or html tags, see https://core.telegram.org/bots/api#markdown-style.
It is configurable if a bot should use formatted messages in the `telegram.cfg`. This pull request is inspired by #5490 but since the other pull request is stale I started a complete rewrite.

### CODE REVIEW

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/TelegramActionService.java [hunk 1]
```diff
@@ -62,8 +62,17 @@ public void updated(Dictionary<String, ?> config) throws ConfigurationException
             for (String bot : bots) {
                 String chatIdKey = String.format("%s.chatId", bot);
                 String tokenKey = String.format("%s.token", bot);
+                String parseKey = String.format("%s.parseMode", bot);
+
                 if (config.get(chatIdKey) != null && config.get(tokenKey) != null) {
-                    Telegram.addToken(bot, (String) config.get(chatIdKey), (String) config.get(tokenKey));
+
+                    String chatId = (String) config.get(chatIdKey);
```
[the remark this answers is not in the document]
Comment (thorstenbe) [reply bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/TelegramActionService.java]: Thank you for your review and your improvements. I fixed these lines of code.

File: bundles/action/org.openhab.action.telegram/README.md [hunk 2]
```diff
@@ -42,11 +42,14 @@ The URL can be specified using the http, https, and file protocols.
...
-|---------------------|---------|:--------:|---------------------------------------|
-| bots                |         | Yes      | Comma-separated list of `<bot-name>`s |
-| `<bot name>.chatId` |         | Yes      | chat id                               |
-| `<bot name>.token`  |         | Yes      | authentication token                  |
+| Property                | Default | Required | Description                                                  |
+|-------------------------|---------|:--------:|--------------------------------------------------------------|
+| bots                    |         | Yes      | Comma-separated list of `<bot-name>`s                        |
+| `<bot name>.chatId`     |         | Yes      | chat id                                                      |
+| `<bot name>.token`      |         | Yes      | authentication token                                         |
+| `<bot name>.parseMode`  |         | No       | Support for formatted messages, values: `Markdown` or `HTML` |
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/README.md]: This needs to specify what the default is when the parameter is not configured.

Suggest something like:

````
Message formatting to be used. Supported values: `Markdown` or `HTML`. Default is Markdown.
````

File: bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/TelegramActionService.java [hunk 3]
```diff
@@ -72,4 +83,8 @@ public void updated(Dictionary<String, ?> config) throws ConfigurationException
             isProperlyConfigured = true;
         }
     }
+
+    private String getConfigValue(Dictionary<String, ?> config, String key) {
```
Comment (9037568) [inline bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/TelegramActionService.java]: This method provides no value and should be removed.

You're also using the Objects.toString method incorrectly, as you've omitted the second `null` parameter.

Comment (thorstenbe) [reply bundles/action/org.openhab.action.telegram/src/main/java/org/openhab/action/telegram/internal/TelegramActionService.java]: This one is fixed. I thought that the default value (if none is given) is `null` but it is `"null"` as String.

File: bundles/action/org.openhab.action.telegram/README.md (hunk 2 above)
Comment (thorstenbe) [reply bundles/action/org.openhab.action.telegram/README.md]: I added a bit more text that hopefully it is better understandable that if no value is set then no formatting is used at all.

**Comment by thorstenbe** [comment]
Do I have to do anything that this pull request gets merged? (Sorry for the dumb question but it is my first one)

**Comment by 9037568** [comment]
Thanks, @thorstenbe !

**Pull request merged**